posts/file_formats_introduction.qmd

```diff
@@ -0,0 +1,220 @@
+
+Your specific needs may require tools with specialized features not universally available. For instance, you might want to store points or regions of interest (ROIs) overlaid on the image, time points for time-series data, or details about transformations applied to the data. An in-depth comparison of these kinds of features is beyond the scope of this introduction, but please see the [metadata features chart](https://datastandards.janelia.org/metadata.html) we have created, which provides a quick comparison of features we know about for the file formats we’ve used.
+
+### Scalability
+
+Scalability is closely related to ease of use. A scalable format is easy to use... at scale. Scalability encompasses storing, viewing, and sharing gigantic images.
+
+Historically, images were limited to non-chunked or 'monolithic' file formats. In these formats, an image is stored as a linear string of bytes in one file. If you want to view even just a piece of your image that is stored in this format, then the entire 2D image must be loaded from your computer’s disk into memory.
+
+The scalability of monolithic formats may be improved by storing multiple copies of the same 2D image at different resolutions in the same file. This type of file is often referred to as a 'multiscale pyramid'. The different resolutions are generated through a technique called downsampling, in which data are typically either uniformly discarded, or in which subsets of pixels are averaged to produce a new, smaller image. Storing an image as a multiscale pyramid means that the software needs to only load into memory the lowest-resolution plane that is needed. However, even though it is a smaller plane, the entire plane must still be loaded into memory.
+
+Another technique for improved scalability is chunking. Chunking involves breaking up an image into chunks. The images can be any number of dimensions, so it's easier to think of these data as n-dimensional arrays rather than 2D grids of pixels. The arrays can be broken up into uniform but otherwise arbitrary chunk sizes. The groups of chunks may be organized as a hierarchy of nested directories on a file system. The metadata, which are usually in a dedicated file, contain information describing the layout of the chunks. Chunked formats can store more complex images than non-chunked formats, and can be viewed more efficiently, because software can retrieve small chunks of data as needed. Chunking and multiscale pyramids are not mutually exclusive; many file formats recommend both. 
```

Comment:
We should motivate the need for chunks.

```suggestion
Another technique for improved scalability is chunking. Chunking involves breaking up an image into local regions called chunks. The images can be any number of dimensions, so it's easier to think of these data as n-dimensional arrays rather than 2D grids of pixels. The arrays can be broken up into uniform, but otherwise arbitrary chunk sizes that reflect the dimensionality of the array. A large 3D volume can be divided into small subvolumes. The groups of chunks may be organized as a hierarchy of nested directories on a file system or web service, making it easy to locate the desired chunk. The metadata, which are usually in a dedicated file, contain information describing the layout of the chunks. Chunked formats can store more complex images than non-chunked formats and can be viewed more efficiently because software can retrieve small chunks of data as needed. Chunking and multiscale pyramids are not mutually exclusive; many file formats recommend both. 
```